### PR TITLE
User can unlike another post

### PR DESCRIPTION
**Pull request by salindholm**
[User can unlike another post](https://www.pivotaltracker.com/story/show/176798583)

## User Story
```
As an API,            
In order to provide the client with the functionality to unlike a post,         
I want to able to have a destroy action.
```

## Changes proposed in this pull request:
* Adding destroy action in likes controller
* Adding route
* Creating test

## What I have learned working on this feature:
* To watch our for false-positive tests

### CODE REVIEW

File: app/controllers/api/likes_controller.rb [hunk 1]
```diff
@@ -1,15 +1,25 @@
...
+
   def create
     if already_liked?
-      render json: { message: "You can't like more than once" }, status: 422
+      destroy
     else
-    like = current_user.likes.create(like_params)
-    render status: 201 if like.persisted?
+      like = current_user.likes.create(like_params)
+      render status: 201
```
Comment (KyunginNa) [inline app/controllers/api/likes_controller.rb]: Why did you delete if like.persisted? :)

File: spec/requests/api/user_can_like_another_post_spec.rb [hunk 2]
```diff
@@ -23,25 +23,25 @@
...
-    it 'is expected to return a 422 status' do
-      expect(response).to have_http_status 422
+    it 'is expected to return a 204 status' do
+      expect(response).to have_http_status 204
     end
 
-    it 'is expected to return an error message' do
-      expect(response_json['message']).to eq "You can't like more than once"
+    it 'is expected to decrease like count back to 0' do
+      expect(Likes.count).to eq 0
```
Comment (KyunginNa) [inline spec/requests/api/user_can_like_another_post_spec.rb]: Isn't it `Like.count`?

**Review by KyunginNa — COMMENTED** [review]
Left some comments!

File: spec/requests/api/user_can_like_another_post_spec.rb (hunk 2 above)
Comment (salindholm) [reply spec/requests/api/user_can_like_another_post_spec.rb]: That was changed in the last commit if you check again :)

File: app/controllers/api/likes_controller.rb (hunk 1 above)
Comment (salindholm) [reply app/controllers/api/likes_controller.rb]: Sorry this was a mistake, it's now back in business!

Comment (KyunginNa) [reply app/controllers/api/likes_controller.rb]: I still can't see it...:(

Comment (rithreaksa) [reply app/controllers/api/likes_controller.rb]: me too :((

Comment (salindholm) [reply app/controllers/api/likes_controller.rb]: Now then..?

Comment (KyunginNa) [reply app/controllers/api/likes_controller.rb]: Now I can see it!

File: app/controllers/api/likes_controller.rb [hunk 3]
```diff
@@ -1,15 +1,25 @@
...
   def create
     if already_liked?
-      render json: { message: "You can't like more than once" }, status: 422
+      destroy
     else
-    like = current_user.likes.create(like_params)
-    render status: 201 if like.persisted?
+      like = current_user.likes.create(like_params)
+      render status: 201 if like.persisted?
     end
```
Comment (rithreaksa) [inline app/controllers/api/likes_controller.rb:10]: yes!! I can see it now :)

File: app/controllers/api/likes_controller.rb [hunk 4]
```diff
@@ -1,15 +1,25 @@
...
+
   def create
     if already_liked?
-      render json: { message: "You can't like more than once" }, status: 422
+      destroy
     else
-    like = current_user.likes.create(like_params)
-    render status: 201 if like.persisted?
+      like = current_user.likes.create(like_params)
+      render status: 201 if like.persisted?
```
Comment (rithreaksa) [inline app/controllers/api/likes_controller.rb:9]: yes i can see it now!!
